metaflow/plugins/kfp/kfp.py

```diff
@@ -642,6 +642,15 @@ def _set_container_labels(
                 "tags.ledger.zgtools.net/ai-experiment-name", self.experiment
             )
 
+        # - In context of Zillow CICD self.username == "cicd_compile"
```

Comment:
Do you know where the `cicd_compile` username is from?

Also potentially do we want to introduce a new label/tag like "source" or something so we can differentiate between CICD, Recurring Run, Ad-hoc local vs Ad-hoc KFNB potentially? Happy to do this outside of this PR if that's easier too just thinking that'd be a nice dimension to aggregate on for usage stats!

Reply:
In our `aip-doit-modules` (I think it should be moved up into CICD) [code link](https://gitlab.zgtools.net/git/analytics/artificial-intelligence/ai-platform/sdks/aip-doit-modules/blob/4340b925ad61efeb3b01d41902e83caaf643fc6c/aip_doit_modules/shared_doit.py#L435).

I'll a add `zodiac.zillowgroup.net/source` that picks up an environment variable that can be set by the environment.